### PR TITLE
Deprecate the provider and proxy to upstream Airflow provider

### PR DESCRIPTION
**Pull request by pankajkoti**
As part of Astronomer's internal plans and decisions, we've decided to contribute the existing functionality provided by the operators and plugins in this repository to the official Apache Airflow Databricks provider. To achieve this, we submitted the following PRs to the Airflow provider:

1. https://github.com/apache/airflow/pull/39178
2. https://github.com/apache/airflow/pull/39771
3. https://github.com/apache/airflow/pull/40013
4. https://github.com/apache/airflow/pull/40724
5. https://github.com/apache/airflow/pull/39295

All functionality has now been contributed to the Airflow Databricks provider, and ongoing support will be maintained there. As a result, we're deprecating the operators and plugins in this repository. Users are encouraged to transition to the official Apache Airflow Databricks provider as soon as possible. The migration process is straightforward—simply update the import path to point to the Airflow provider and ensure that you install `apache-airflow-providers-databricks>=6.8.0`, which includes all the contributions mentioned above.

closes: https://github.com/astronomer/issues-airflow/issues/715

### CODE REVIEW

File: pyproject.toml
Comment (pankajkoti) [inline pyproject.toml:39]: Wait for the provider release of 6.8.0 before merging this PR

```suggestion
    "apache-airflow-providers-databricks>=6.8.0",
```

File: README.md
Comment (pankajastro) [inline README.md:10]:
```suggestion
| <code>`from astro_databricks.operators.notebook import DatabricksNotebookOperator`</code>  | <code>`from airflow.providers.databricks.operators.databricks import DatabricksNotebookOperator`</code>                  |
```
Shall we verify if this works and see how it renders?

**Review by pankajastro — APPROVED** [review]
LGTM

File: README.md
Comment (pankajkoti) [reply README.md:10]: The current text renders fine if we look at the rich diff on the PR review files page

Screenshot 2024-08-07 at 11 44 18 AM

**Pull request merged**
